Remotion/Data/DomainObjects.UnitTests/TestDomainDatabase.cs

```diff
@@ -0,0 +1,103 @@
+    private static StandardMappingDatabaseAgent s_standardMappingDatabaseAgent;
+
+    public static void InitializeDatabase ()
+    {
+      if (s_initialized)
+        return;
+
+      lock (s_lock)
+      {
+        if (s_initialized)
+          return;
+
+        s_initialized = true;
+
+        try
+        {
+          StandardConfiguration.EnsureInitialized();
+          TableInheritanceConfiguration.EnsureInitialized();
+
+          SqlConnection.ClearAllPools();
+
+          var masterAgent = new DatabaseAgent(DatabaseTest.MasterConnectionString);
+          masterAgent.ExecuteBatchFile("Database\\DataDomainObjects_CreateDB.sql", false, DatabaseConfiguration.GetReplacementDictionary());
+          var testDomainAgent = new DatabaseAgent(DatabaseTest.TestDomainConnectionString);
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TestDomain_TearDownDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TableInheritanceTestDomain_TearDownDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TestDomain_SetupDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_TableInheritanceTestDomain_SetupDB.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+          testDomainAgent.ExecuteBatchFile("Database\\DataDomainObjects_Patches.sql", true, DatabaseConfiguration.GetReplacementDictionary());
+
+          s_standardMappingDatabaseAgent = new StandardMappingDatabaseAgent(DatabaseTest.TestDomainConnectionString);
+          string sqlFileName = StandardMappingTest.CreateTestDataFileName;
+          s_standardMappingDatabaseAgent.ExecuteBatchFile(sqlFileName, true, DatabaseConfiguration.GetReplacementDictionary());
+          string sqlFileName1 = TableInheritanceMappingTest.CreateTestDataFileName;
+          s_standardMappingDatabaseAgent.ExecuteBatchFile(sqlFileName1, true, DatabaseConfiguration.GetReplacementDictionary());
+
+          // We don't want the tests to initialize a default mapping; therefore, modify MappingConfiguration.s_fields.Current so that it will 
+          // throw when asked to generate a new MappingConfiguration.
+
+          var throwingMappingConfigurationContainer = new DoubleCheckedLockingContainer<IMappingConfiguration>(
+              () =>
+              {
+                throw new InvalidOperationException(
+                    "This test failed to setup the mapping configuration. Did you forget to derive from StandardMappingTest or to call base.SetUp?");
+              });
+          var fields = PrivateInvoke.GetNonPublicStaticField(typeof(MappingConfiguration), "s_fields");
+          Assertion.IsNotNull(fields);
+          PrivateInvoke.SetPublicField(fields, "Current", throwingMappingConfigurationContainer);
```

Comment:
the missing backup / restore logic we discussed earlier: yes, the SetupFixture was in the root and thus the restore didn't do anything. Now that we have a helper-api, I think the missing reset of the original mapping config is a problem if you start to use the helper in a more granular fashion.  

Alternative:
if you consider that this entire helper type should only be used once to initialize the database, we can actually drop the CleanupDatabase method because we don't need to clear the sql connections at the end of the testrun. the process ends anyway.

Reply:
The cleanup is still located in the `SetupFixture.OneTimeTearDown` so this code is not necessary. It would be better if we could cleanup the DB right after all the tests that needed it ran but this is not possible in NUnit.